docs/cram-tests/index.md

```diff
@@ -0,0 +1,368 @@
+  $ node ./output/src/order-confirmation/SandwichTests.mjs
+```
+
+Note that this is the same path we specified inside the `cram/deps` field.
+
+To run tests, execute
+
+```shell
+opam exec -- dune build @runtest
+```
+
+Here,
+[@runtest](https://dune.readthedocs.io/en/stable/reference/aliases.html#runtest)
+is a built-in *alias*. The above command builds whatever is needed by the tests
+and then runs the tests.
+
+## Aliases
+
+A Dune [alias](https://dune.readthedocs.io/en/stable/reference/aliases.html) is
+a virtual build target that other build targets can attach to. You can refer to
+aliases by name in build commands (like the one above), to build a specific
+subset of your build targets. In commands, we need to put `@` in front of the
+alias name to distinguish it from file names. Therefore when we refer to an
+alias, we put an `@` in front, e.g. `@runtest`, `@melange`, etc.
+
+::: tip
+
+Cram tests are automatically attached to the `@runtest` alias, and
+`melange.emit` stanzas are automatically attached to the `@melange` alias.
+
+:::
+
+## Promotion
+
+After running `opam exec -- dune build @runtest`, you should be able to see the
+output from the cram test you just added. However, this doesn't count as a
+passing cram test, because this output can't be compared to the expected output.
+Dune has a nice feature that lets you
+[promote](https://dune.readthedocs.io/en/stable/concepts/promotion.html#promotion)
+the latest output of a cram test to become the expected output:
+
+```shell
+opam exec -- dune promote
+```
+
+The above command updates your `src/order-confirmation/tests.t` cram test file,
+which now looks something like this:
+
+<<< tests{cram}
+
+The part underneath the command is the expected output of the command.
+
+## Add `npm` scripts
+
+Instead of typing `opam exec` commands over and over, let's add the relevant
+`npm` scripts to your `package.json` file:
+
+```json
+"test": "npm run build -- @runtest",
+"testwatch": "npm run build -- @runtest --watch",
+"promote": "npm run dune -- promote"
+```
+
+You should also add some lines to the `scriptsComments` section of
+`package.json`:
+
+```json
+"test": "# Run the tests",
+"testwatch": "# Watch files and re-run tests",
+"promote": "# Promote most recent output to expected output"
+```
+
+Run `npm run test` to check that it works.
+
+## Sanitize cram test output
+
+The cram test still doesn't succeed, but now it's because the most recent output
+and the expected output don't match:
+
+<<< diff{diff}
+
+The problem is that the recorded durations for individual unit tests is differs
+on each test run. Therefore we must
+[sanitize](https://dune.readthedocs.io/en/stable/tests.html#test-output-sanitation)
+the test output, i.e. remove the parts that are non-deterministic. Update your
+cram test command:
+
+```shell
+$ node ./output/src/order-confirmation/SandwichTests.mjs | sed '/duration_ms/d'
+```
+
+Now the output of the command is piped to `sed '/duration_ms/d'`, which removes
```

Comment:
I'm using `sed` here, which doesn't work on Windows. Not sure if it's going to be a problem later on.

Reply:
Sounds great!